config/v1/types_ingress.go

```diff
@@ -82,8 +82,52 @@ type IngressSpec struct {
 	// Note that if there are no RequiredHSTSPolicies, any HSTS Policy annotation on the route is valid.
 	// +optional
 	RequiredHSTSPolicies []RequiredHSTSPolicy `json:"requiredHSTSPolicies,omitempty"`
+
+	// platform holds desired information specific to the underlying
+	// infrastructure provider for the ingress.
+	Platform *IngressPlatformSpec `json:"platform,omitempty"`
+}
+
+type IngressPlatformSpec struct {
+	// type is the underlying infrastructure provider for the cluster.
+	// Allowed values are "AWS", "Azure", "BareMetal", "GCP", "Libvirt",
+	// "OpenStack", "VSphere", "oVirt", "KubeVirt", "EquinixMetal", "PowerVS",
+	// "AlibabaCloud", "Nutanix" and "None". Individual components may not support all platforms,
+	// and must handle unrecognized platforms as None if they do not support that platform.
+	//
+	// +unionDiscriminator
+	Type PlatformType `json:"type"`
+
+	// AWS contains settings specific to the Amazon Web Services infrastructure provider.
+	// +optional
+	AWS *AWSIngressSpec `json:"aws,omitempty"`
+}
+
+type AWSIngressSpec struct {
+	// LoadBalancer contains the AWS load balancer details
+	LoadBalancer *LoadBalancer `json:"loadbalancer,omitempty"`
```

Comment:
Is this optional? If so it needs an `+optional` tag

Also, please don't use a pointer, it harms discoverability for the end user

Reply:
But, how can we check for nil values in the operator code if we don't use a pointer ?
Eg- Check if Spec.Platform.LoadBalancer != nil if it is not set as it is an optional field

Reply:
Why do you need to check if it is nil? Do you need to distinguish between a nil value and the zero value of the struct?

Reply:
I just want to check if LoadBalancer is set or not. From my understanding Go will throw nil error if this value is not set as it is optional.